src/synthesizer/emission_models/transformers/dust_attenuation.py

```
@@ -332,38 +332,44 @@ def N09Tau(lam, slope, cent_lam, ampl, gamma):
         )
         func = interpolate.interp1d(
-            lam.value[ok1], k_lam[ok1], fill_value="extrapolate"
+            _lam.value[ok1], k_lam[ok1], fill_value="extrapolate"
         )
     if np.sum(ok2) > 0:  # equation 2
-        k_lam[ok2] = -1.857 + (1.040 / lam.value[ok2])
+        k_lam[ok2] = -1.857 + (1.040 / _lam.value[ok2])
     if np.sum(ok3) > 0:
         # Extrapolating the 0.12um<=lam<0.63um regime
-        k_lam[ok3] = func(lam.value[ok3])
+        k_lam[ok3] = func(_lam.value[ok3])
```

Pylint (3.3.7)

[error] Possibly using variable 'func' before assignment

(E0606)